### ISSUE TITLE
Combo Box

### ISSUE THREAD

**Issue by joshwooding** [jpmorganchase/salt-ds#2763]
We need to re-visit Combo Box to add missing features and update its API so it matches Salt's API goals.

```[tasklist]
### Tasks
- [ ] #2406
- [ ] #2459
- [ ] #2410
- [ ] #2411
- [ ] #2413
- [x] Update Combo Box to use declarative children
```

**Comment by joshwooding** [comment]
Copied from #2607

20 Nov - snapshot release last week, waiting for MTK feedback, waiting for combobox design spec

**Comment by origami-z** [comment]
22 Nov - Adding some examples to the WIP Combobox and Dropdown and currently writing tests

**Comment by joshwooding** [comment]
24 Nov - Adding more tests and correcting some small behaviours. 
Fixed Button aria 
Fixed List not showing on click
Fixed Button not having Form Field support
Fixed alt + arrow down or up moving focus to a list item

**Comment by joshwooding** [comment]
27 Nov - Still writing tests and working on it. Still need specs

**Comment by joshwooding** [comment]
1 Dec - Not been able to work on this much. Still need specs but pushed up the work.

**Comment by origami-z** [comment]
8 Dec - PR with feature, test, docs ready for review. Fixing a minor bug around clearing input / selection

**Comment by origami-z** [comment]
11 Dec - waiting for review. Small exampled  needed for UITK behavior

**Comment by joshwooding** [comment]
18 Dec - Call with stakeholders tomorrow

**Comment by joshwooding** [comment]
09/01 - Working on finishing addressing the feedback, have been raising separate PRs to fix issues and had to update floating-ui to solve some issues.

**Comment by joshwooding** [comment]
Closed by #3131
